src/v/io/tests/persistence_test.cc

```diff
@@ -274,6 +274,47 @@ TYPED_TEST(PersistenceTest, ReadWrite) {
     }
 }
 
+class fault_injection : std::exception {};
+
+TYPED_TEST(PersistenceTest, ThrowCreate) {
+    this->fs.fail_next_create(std::make_exception_ptr(fault_injection()));
```

Comment:
nit: there is a fair amount of noise with the `std::make_exception_ptr(fault_injection())`. Can we either:

Consolidate these into a helper function or create a helper method to call `make_exeception_ptr` like seastar does in a couple of places? Ex: https://github.com/scylladb/seastar/blob/4b01caf666187ebae8344b8a0b6b73974b0876d9/include/seastar/core/abort_source.hh#L172-L175